packages/shared/src/mock/utils.ts

```diff
@@ -31,22 +38,27 @@ export function assertMockAttribute<S extends Shape, T extends DType>(
   }
 }
 
-export function findMockEntity(path: string): Entity | undefined {
+export function findMockEntity(path: string): ProvidedEntity | undefined {
   assertAbsolutePath(path);
 
   if (path === '/') {
     return mockMetadata;
   }
 
-  const pathSegments = path.slice(1).split('/');
-  return pathSegments.reduce<Entity | undefined>(
-    (parentEntity, currSegment) => {
-      return parentEntity && isGroup(parentEntity) && hasChildren(parentEntity)
-        ? getChildEntity(parentEntity, currSegment)
-        : undefined;
-    },
-    mockMetadata
-  );
+  const parentPath = path.slice(0, path.lastIndexOf('/')) || '/';
+  const parent = findMockEntity(parentPath);
+  if (!parent) {
+    return undefined;
+  }
+
+  assertGroup(parent);
+  const childName = path.slice(path.lastIndexOf('/') + 1);
+  const child = parent && getChildEntity(parent, childName);
+
+  if (child && isGroup(child)) {
+    assertGroupWithChildren(child);
```

Comment:
`getChildEntity` has no way to know that it's dealing with mock metadata where all groups have children, so we have to assert. This is the only place left where we need to call `assertGroupWithChildren`, but I've simplified it to accept a group instead of an entity, so it's now only checking for the presence of `children`.